packages/frontend/app/register/weekly-meetings/components/location-search.tsx

```diff
@@ -0,0 +1,163 @@
+"use client";
+
+import React from "react";
+
+import { Button } from "@nextui-org/button";
+import { Card, CardHeader, CardBody, CardFooter } from "@nextui-org/card";
+import { Spinner } from "@nextui-org/spinner";
+import axios from "axios";
+import { motion } from "framer-motion";
+import { Locate } from "lucide-react";
+import { useForm } from "react-hook-form";
+import { useDispatch, useSelector } from "react-redux";
+
+import { Button as ShadButton } from "@/components/ui/button";
+import {
+  Form,
+  FormField,
+  FormItem,
+  FormControl,
+  FormDescription,
+  FormLabel,
+  FormMessage,
+} from "@/components/ui/form";
+import { Input } from "@/components/ui/input";
+import { ScrollArea, ScrollBar } from "@/components/ui/scroll-area";
+import {
+  Tooltip,
+  TooltipContent,
+  TooltipProvider,
+  TooltipTrigger,
+} from "@/components/ui/tooltip";
+import { backendRoutes } from "@/lib/config";
+import { AppDispatch, RootState } from "@/lib/stores/app-store";
+import { MeetingsState, meetingsSlice } from "@/lib/stores/local-meetings";
+import { fetchMeetingsThunk } from "@/lib/stores/thunks/fetch-meetings";
+import { LocationSearchFormData } from "@/lib/types/location-form";
+
+const { updateUserCoords } = meetingsSlice.actions;
+
+export function LocationSearch({
+  setUseCurrentLocation,
+}: {
+  setUseCurrentLocation: React.Dispatch<React.SetStateAction<boolean>>;
+}) {
+  const form = useForm<LocationSearchFormData>();
+  const [locationData, setLocationData] = React.useState<any | never>([]);
+  const [isLoading, setIsLoading] = React.useState<boolean>(false);
+  const dispatch: AppDispatch = useDispatch();
+  const state: MeetingsState = useSelector(
+    (state: RootState) => state.meetings,
+  );
+
+  const onSubmit = async (): Promise<void> => {
+    setLocationData([]);
+    setIsLoading(true);
+    const geoCodingRes = await axios.get(
+      `${backendRoutes.user.findLocation}?q=${form.getValues().query}`,
+      { headers: { Authorization: sessionStorage.getItem("sessionToken") } },
+    );
+
+    setLocationData([...geoCodingRes.data.locations]);
+    setIsLoading(false);
+  };
```

Comment:
All of this should really be done in the redux state.

Reply:
Hmmm I would understand if this data was being used in multiple components across the project, but for simple stuff like this I feel like React state is good enough.

Reply:
It's true, it's good enough. However implementing redux is an abstraction. We decouple the request-based state management from the component and bridge them with a thunk.